### DISCUSSION TITLE
Importing existing scenes

### DISCUSSION THREAD

**Discussion by javageek68** [carla-simulator/carla#7187] [Q&A]
Is there an established workflow for importing existing maps?  I have a very complex map in another Unreal project that I would like to import into Carla.  This is an off-road scenario.  So I don't need to create roads with Road Runner.  And, I don't have the option of using Road Runner (or TrueVision) anyway.  I don't have enough time before my deadline to rebuild the map from scratch.  The only choice I have is to import the existing one and make it work with Carla.

I imported the map along with its terrain, materials and props as a plug in.  I dropped in a BP_Sky blueprint.  But when I ran the project, the server did not start.  Are there any other blueprints I need to add to my map to make it work with Carla?

**Comment by ahlamiki** [comment]
Hi @javageek68 
First of all, we apologize for the delay in responding to the issues. We have been investing the majority of our resources in other tasks. However, we will now return to providing more support to the community. We apologize for any inconvenience this may have caused.

Provably, you have problems running the project in Carla because you don't have the .xodr information. On this link, you can follow alternative methods to import maps with RoadRunner, you have the manual import option as well:
https://carla.readthedocs.io/en/latest/tuto_M_add_map_alternative/#import-map

**Comment by javageek68** [comment]
Thanks for the reply.

I don't have Roadrunner.  Since the download site has a button called "Request a quote" instead of the actual price, I assumed that it was insanely expensive.   I work at a school.  Do you know if there is a way to get an academic license?

I downloaded TrueVision.  Unfortunately, it seemed extremely limited.  My project required me using the heightmap of a real-world area to recreate the terrain.  I didn't see a way to import the heightmap into it TrueVision.  Does RoadRunner have that feature?

Thanks,

Mike
________________________________

**Reply by bitvijays** [reply]
@javageek68 Hope you are doing well. Just curious, the truevision only exports .xodr file. However, carla requires .fbx and .xodr file. Was there a way you were able to import the map into Carla with just .xodr file?

**Reply by lakhanqurban** [reply]
> @javageek68 Hope you are doing well. Just curious, the truevision only exports .xodr file. However, carla requires .fbx and .xodr file. Was there a way you were able to import the map into Carla with just .xodr file?

Hi, have you found any solution to this? as I also want to import a (.xodr) file that only has road (not detailed map).

**Comment by ahlamiki** [comment]
Hi @javageek68

Yes, RoadRunner has the feature of importing heightmaps. Regarding the license, you can check their website or even the forums, where you can find information about the type of license they are offering: https://es.mathworks.com/products/roadrunner.html.

**Comment by javageek68** [comment]
I imported a minimal xodr file just to get the scene started.  I had no idea what other assets had to be added to a scene for Carla to accept it.  So, I did that.

Then, I imported the heightmap for my scene into Unreal, built it the regular way, and deleted the terrain from the xodr file.

It mostly works.  But that brings up a question.  When I run the scene, I can use the Python API to query the spawn points in my custom scene and that works fine.  But when I query the waypoints I always get an empty list.  I've got no idea what Carla is looking for.  Is there a way I can troubleshoot that?

Thanks again,

Mike
________________________________

**Comment by ahlamiki** [comment]
Hi @javageek68,

We apologize for the delay in responding to the issues. If you are using just an xodr file it has to be working, could you give me more details regarding this problem? Maybe a screenshot of what you get when you query the waypoints?
